torch/_inductor/codegen/cuda/cuda_cpp_scheduling.py

```diff
@@ -40,9 +45,32 @@ def is_cuda_cpp_template(node: BaseSchedulerNode) -> bool:
             node.node, CUDATemplateBuffer
         )
 
+    def is_cuda_cpp_fused_template(self, node: BaseSchedulerNode) -> bool:
+        return isinstance(node, FusedSchedulerNode) and self.is_cuda_cpp_template(node)
+
     def can_fuse_vertical(
         self, node1: BaseSchedulerNode, node2: BaseSchedulerNode
     ) -> bool:
+        if self.is_cuda_cpp_template(node1) and isinstance(node2, SchedulerNode):
+            assert node1.node, "node1.node should not be None"
+            assert node2.node, "node2.node should not be None"
+            return self._can_fuse_epilogue_impl(
+                cast(CUDATemplateBuffer, node1.node),
+                [],
+                node2,  # type: ignore[arg-type]
+            )
+        elif self.is_cuda_cpp_fused_template(node1) and isinstance(
```

Comment:
Typically we fuse all of the pointwise nodes first. Then fuse into templates. So its more likely that the second node is a FusedSchedulerNode and the cpp template is still just `cuda_cpp_template`..